Provide strategy for updating Prow cluster (#148) (#747)
kyma-project/test-infra: docs/prow/prow-cluster-update.md

## Update process

To update a Prow cluster follow these steps:

1. Follow [this](./prow-installation-on-forks.md) document for details to set up a Prow cluster.
2. Go to the [`kubernetes/test-infra`](https://github.com/kubernetes/test-infra/) project and select a commit with the desired update for the Prow cluster. For example, use [2c8e0dbb96b4c1a86d42275dfbed5474a6d05def](https://github.com/kubernetes/test-infra/commit/2c8e0dbb96b4c1a86d42275dfbed5474a6d05def).
3. Open both [`starter.yaml`](../../prow/cluster/starter.yaml) in the current project and [`starter.yaml`](https://github.com/kubernetes/test-infra/blob/2c8e0dbb96b4c1a86d42275dfbed5474a6d05def/prow/cluster/starter.yaml) in the Kubernetes project and copy new tags for these containers:
    * gcr.io/k8s-prow/hook
    * gcr.io/k8s-prow/plank
    * gcr.io/k8s-prow/sinker
    * gcr.io/k8s-prow/deck
    * gcr.io/k8s-prow/horologium
    * gcr.io/k8s-prow/tide
4. Copy the commit ID into a comment at the top of the file to keep track of the current release used for the deployments.

> 5. Check with your preferred diff tool for other meaningful changes (e.g. stability update in a prow component or additional configurations on existing components) that need to be taken into account and copied over.

Reviewer: I'm still not sure about the word "meaningful". Who decides if a change is meaningful or not? It is unclear for a reader what they should copy. Should they look for more changes between the `starter.yaml` files or check other files? Is 7th step of the instruction a duplication or does it refer to the `config.yaml` files?

Author: I'd assume that SIG-Deploy-CI decides on meaningful. So far, the upgrade progress includes only starter.yaml and config.yaml.

5th step applies to starter.yaml, 7th step applies to config.yaml

Reviewer: ```suggestion
5. Use your preferred diff tool to check and copy the stability update in a Prow component and additional configurations on existing components.
```

6. Open both [`config.yaml`](../../prow/config.yaml) in the current project and [`config.yaml`](https://github.com/kubernetes/test-infra/blob/2c8e0dbb96b4c1a86d42275dfbed5474a6d05def/prow/config.yaml) in the Kubernetes project and copy new tags for these containers:
    * gcr.io/k8s-prow/clonerefs
    * gcr.io/k8s-prow/initupload
    * gcr.io/k8s-prow/entrypoint
    * gcr.io/k8s-prow/sidecar

> 7. Check with your preferred diff tool for other meaningful changes (e.g. stability update in a prow component or additional configurations on existing components) that need to be taken into account and copied over.

Reviewer: ```suggestion
7. Use your preferred diff tool to check and copy the stability update in a Prow component and additional configurations on existing components.
```
